src/os/bluestore/BlueStore.cc

```diff
@@ -8097,7 +8097,18 @@ BlueStore::OnodeRef BlueStore::fsck_check_objects_shallow(
   if (!o->extent_map.shards.empty()) {
     ++num_sharded_objects;
     int64_t shard_errors = 0;
-    // we should have all the shards we expect to have, in order
+    // we should have all the shards we expect to have
```

Comment:
Can we avoid twin loop over extent_map.shards and set missing_shard flag in the loop below?  This will cause extent_map to be partially filled with shard_info if some missed shard is found in the mid but that's probably fine. 

I think this is a better approach given that "missed shard" is a very rare case while duplicate enumerating over shards container is a common one (every onode scan will suffer from that).

Or I'm missing something?